lib/matrix.js

```diff
@@ -52,11 +58,351 @@ var init = function(exports){
     var HEADERS = {
         "User-Agent": "matrix-js"
     };
+    
+    // Basic DAOs to abstract slightly from the line protocol and let the
+    // application customise events with domain-specific info
+    // (e.g. chat-specific semantics) if it so desires.
+    
+    /*
+     * Construct a Matrix Event object
+     * @param {Object} event The raw event to be wrapped in this DAO
+     */
+    function MatrixEvent(event) {
+        this.event = event;
```

Comment:
You should probably be setting ``{ }`` as the default here else any of the ``get`` functions will throw.